### ISSUE TITLE
Bug on enter key on iOS

### ISSUE THREAD

**Issue by largeden** [froala/wysiwyg-editor#2007]
##### Steps to reproduce the problem.
![ios_enter_bug1](https://cloud.githubusercontent.com/assets/5575931/26630775/3379a130-4643-11e7-8b32-0a7993dfc58a.gif)

##### OS.
IOS iphone, ipad

##### froala editor version
2.6.1

---
I found a bug I had for a long time.
At least this bug did not happen until version 2.0.1.
I suspect the function below.

```javascript
//froala_editor.js
  $.FE.MODULES.keys = function (editor) {
    var IME = false;

    /**
     * ENTER.
     */
    function _enter (e) {
      if (!editor.opts.multiLine) {
        e.preventDefault();
        e.stopPropagation();
      }
      else if (!editor.helpers.isIOS()) {
        e.preventDefault();
        e.stopPropagation();

        if (!editor.selection.isCollapsed()) editor.selection.remove();

        editor.cursor.enter();
      }
    }
```

I thought it was cause `isIOS()`, so I removed the code and the problem was resolved.

![ios_enter_bug2](https://cloud.githubusercontent.com/assets/5575931/26631309/6008ea1a-4645-11e7-9408-9447304bbf86.gif)

However, there was a more important issue. It is a Korean input problem.
I found that running `editor.cursor.enter()` on IOS caused Korean problems.
And I've concluded that the cause of this problem is in `_mapKeyDown()`

![ios_enter_bug3](https://cloud.githubusercontent.com/assets/5575931/26631396/c2b36d5c-4645-11e7-91d7-cad465f15ea2.gif)

Again, this problem does not occur in v2.0.1.
In other words, changing `_enter(e)`, `_mapKeyDown(e)` from v2.6.1 to v2.0.1 fixes the bug.

Please fix the problem.

**Comment by stefanneculai** [comment]
@largeden I am not sure I understand exactly how to reproduce this? Could you please offer few more details? Maybe a step by step? Thanks a lot!

**Comment by largeden** [comment]
@stefanneculai 
I'm sorry I did not explain.

![ios_enter_bug1](https://cloud.githubusercontent.com/assets/5575931/26630775/3379a130-4643-11e7-8b32-0a7993dfc58a.gif)

(The default paragraph is the P tag)
#### Step1.
P Write the text in the paragraph and replace it with H1.
#### Step2.
Press Enter.
The div tag is displayed instead of the default paragraph P.
(On the macbook, it appears as the default Paragraph P tag.)

The next video is talking about a solution and a new bug.
But please forget it now. :)

**Comment by stefanneculai** [comment]
@largeden could you please try with the current master from Github and let us know if the problem is fixed?

**Comment by largeden** [comment]
@stefanneculai 
Thank you! =b
I think that the problem that I pointed out at first is solved.
![bugfix](https://user-images.githubusercontent.com/5575931/27290139-4d1df2f4-5547-11e7-966d-e411ca04517a.gif)

However... if you press Enter after entering Korean, two lines may be created.
This is very rare and patterns are hard to find.
There is a feeling that it is canceled from the first word after entering the enter key.
(At least it was not in v2.0.1)

I think this issue has been resolved. close please.
When the problem of Hangul becomes clear, I will register the issue again.

**Comment by largeden** [comment]
@stefanneculai 
oh..
![new_bug](https://user-images.githubusercontent.com/5575931/27291121-5579c01a-554a-11e7-8654-489bb2040e02.gif)
After typing dot (.) And pressing enter, there are two lines..

**Comment by stefanneculai** [comment]
@largeden could you please check it again now?

**Comment by largeden** [comment]
@stefanneculai 👍 
all clear!
Korean, Japanese, special characters, and English typing also works normally.
Thank you. :)

**Comment by stefanneculai** [comment]
Thanks for helping us to nail it 😃 .
